README.md

```diff
@@ -1,60 +1,82 @@
+
+We recommend using [CMake Package Manager (CPM)](https://github.com/TheLartians/CPM.cmake) to fetch `cuCollections` into your project.
+With CPM, getting `cuCollections` is easy:
+
 ```
+cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
+
+include(path/to/CPM.cmake)
+
+CPMAddPackage(
+  NAME cuco
+  GITHUB_REPOSITORY NVIDIA/cuCollections
+  GIT_TAG dev
+  OPTIONS
+     "BUILD_TESTS OFF"
+     "BUILD_BENCHMARKS OFF"
+     "BUILD_EXAMPLES OFF"
+)
 
-## Quick Start
-[Building](#building-cuCollections) the project as instructed, will build the examples under `build/examples` path. Examples are designed to be simple and standalone. In what follows, the most basic map operations (i.e., insertion and find) is detailed. The corresponding program can be found in `$cuCollectionsPath/examples/map/insert_find.cu`.
-```cpp
-int main() {
-  // Synthesizing dummy data to put in hashmap
-  std::vector<int> keys(N);
-  std::vector<double> vals(N);
-  for (unsigned int i = 0; i < N; i++) {
-    keys[i] = i;
-    vals[i] = i * 0.5;
-  }
-
-  cuCollections::unordered_map<int, double> map(N);
-  map.insert(keys, vals);
-  auto results = map.find(keys);
-
-  for (unsigned int i = 0; i < results.size(); i++) {
-    std::cout << keys[i] << ": " << results[i] << std::endl;
-  }
-}
+target_link_libraries(my_library cuco)
 ```
-Using a GPU-based map can be very beneficial when the workload includes some parallelism. As a result, `cuCollections::unordered_map` will accept, vectors of keys and values and insert them concurrently. This code snippet synthesizes dummy vectors for keys and values, then it inserts them in a `cuCollections::unordered_map`. Subsequently, it queries the map and prints the results which are returned in a vector called `results`. You may refer to the `examples` directory for more usage instances.
-
-## Integration
-`cuCollections` is designed to be added to projects as a submodule. In such settings, examples and benchmarks will not be build. As instructed in what follows, it is straightforward to integrate cuCollections in an existing project.
-- Add `cuCollections` as submodule: 
-```bash
-mkdir thirdparty && cd thirdparty
-git submodule add https://github.com/rapidsai/cuCollections.git cuCollections
-git submodule update --init --recursive
+
+This will take care of downloading `cuCollections` from GitHub and making the headers available in a location that can be found by CMake. Linking against the `cuco` target will provide everything needed for `cuco` to be used by the `my_library` target.
+
+[1] `cuCollections` is header-only and therefore there is no binary component to "link" against. The linking terminology comes from CMake's `target_link_libraries` which is still used even for header-only library targets. 
+
+## Dependencies
+- C++14
+- [libcu++](https://github.com/NVIDIA/libcudacxx)
+- [CUB](https://github.com/thrust/cub)
```

Comment:
A few questions:
- Doesn't CUB come with CUDA Toolkit since 11.0 version? 
- Also should we add which CUDA Toolkit Version XX is required?
- When we say `libcu++` is a dependency, does that mean users need to go get it somehow separately from `cuco`? We should clarify what dependencies mean, something you need to install up front or something that comes with the cuco installation.

Reply:
Nothing needs to be fetched by the user. The CMake script takes care of downloading all of the dependencies. This includes CUB and libcu++. 

Strictly speaking, I don't know what CUDA toolkit version would be required. We don't depend on anything special being available in the toolkit (like CUB or libcu++).